packages/solutions/module-tools/src/builder/esbuild/adapter.ts

```diff
@@ -212,7 +212,7 @@ export const adapterPlugin = (compiler: ICompiler): Plugin => {
         if (args.namespace === globalNamespace) {
           const value = config.umdGlobals[args.path];
           return {
-            contents: `module.exports = (typeof globalThis !== "undefined" ? globalThis : Function('return this')() || global || self)[${JSON.stringify(
+            contents: `module.exports = (typeof globalThis !== "undefined" ? globalThis : global || self || Function('return this')())[${JSON.stringify(
```

Comment:
Can you provide some context for this change?

Reply:
You can read about `global` and `self` in caniuse.

Reply:
@targeral Will this change break the code in the browsers?

![image](https://github.com/web-infra-dev/modern.js/assets/7237365/250d6c96-0fe1-40ad-873e-255cf3df66b6)

Reply:
OK, I will fix it.

Reply:
You can review again.

Reply:
LGTM 👍

Reply:
Thx